### ISSUE TITLE
astrometry-net: image2pnm fails (because of Python path)

### ISSUE THREAD

**Issue by dstndstn** [Homebrew/homebrew-core#54256]
# Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

image2pnm

To test that it works.

## What happened (include command output)

```
> which image2pnm
/usr/local/bin/image2pnm
> image2pnm
Traceback (most recent call last):
  File "/usr/local/bin/image2pnm", line 14, in <module>
    from astrometry.util.image2pnm import main
ModuleNotFoundError: No module named 'astrometry'
```

## What you expected to happen

Print help message

## Step-by-step reproduction instructions (by running `brew install` commands)

brew install astrometry-net
cd /tmp
unset PYTHONPATH
image2pnm

**Comment by dstndstn** [comment]
I'm the upstream author, but others have developed the homebrew formula.

Not sure what happened here -- bin/image2pnm is a python script that uses python code in the astrometry-net formula.

It looks like the formula is using a virtualenv, but not pointing the image2pnm (and other scripts) to use that virtualenv -- instead it's pointing to the system-wide /usr/local/opt/python@3.8/bin/python3 python.

**Comment by dstndstn** [comment]
I think you want to set PYTHON_SCRIPT to the virtualenv python

**Comment by dstndstn** [comment]
Note that the astrometry.net Makefile rewrites the python shebang if you set PYTHON_SCRIPT, but since the formula already has a shebang-rewriting, it shoots the Makefile in the foot

**Comment by SMillerDev** [comment]
https://www.python.org/dev/peps/pep-0394/#recommendation says Homebrew should rewrite the shebang to something specific. Is `PYTHON_SCRIPT` an established standard?

**Comment by dstndstn** [comment]
Not that I know of -- it's specifically in the Makefile so that people can specify exactly which python script they want to have in the shebang.

I have a fix and am sending a PR, give me a minute.

**Comment by dstndstn** [comment]
PR #54258

**Comment by iMichka** [comment]
Fixed by #54258! Thanks for the improvement!
